src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -42,17 +42,11 @@ internal interface IPartition<TElement> : IIListProvider<TElement>
 
         IPartition<TElement> Take(int count);
 
-        TElement ElementAt(int index);
+        bool TryGetElementAt(int index, out TElement element);
```

Comment:
On it's own this change is a mixed bag to performance, but it allows for chaining of the operation.